lib/auth/grpcserver.go

```diff
@@ -1330,14 +1330,18 @@ func (g *GRPCServer) UpsertApplicationServer(ctx context.Context, req *proto.Ups
 	server := req.GetServer()
 	app := server.GetApp()
 
-	// Only allow app servers with Okta origins if coming from an Okta role. App servers sourced from
+	// Only allow app servers with Okta subkinds/origins if coming from an Okta role. App servers sourced from
 	// Okta are redirected differently which could create unpredictable or insecure behavior if applied
 	// to non-Okta apps.
```

Comment:
What happens if a proxy that doesn't know about the okta subkinds or the okta origin labels ends up redirecting for such an app?